services/spar/src/Spar/Types.hs

```diff
@@ -143,6 +147,25 @@ instance ToJSON IdPMetadataInfo where
 newtype ScimToken = ScimToken {fromScimToken :: Text}
   deriving (Eq, Show, FromJSON, ToJSON, FromByteString, ToByteString)
 
+newtype ScimTokenHash = ScimTokenHash {fromScimTokenHash :: Text}
+  deriving (Eq, Show)
+
+instance FromByteString ScimTokenHash where
+  parser = string "sha512:" *> (ScimTokenHash <$> parser)
+
+instance ToByteString ScimTokenHash where
+  builder (ScimTokenHash t) = BB.fromByteString "sha512:" <> builder t
```

Comment:
Add a roundtrip test, like [these](https://github.com/wireapp/wire-server/blob/3af1aa63a9cb4d6594a4e017b0ba38408c0f26d2/libs/wire-api/test/unit/Test/Wire/API/Roundtrip/ByteString.hs)?

(I admit that there are more gaps in the roundtrip test coverage, but we might as well start now to patch them...)

Reply:
(we can also do that in a separate PR)

Reply:
added roundtrip test